### PR TITLE
Feature/dwr 397

### PR DESCRIPTION
**Pull request by agorina**
For the reporting side we want to have as much data easily available without extra logic.  This adds the following fields to the item table in reporting:
- primary claim code (in addition to the id)
- primary target code (in addition to the id)
- difficulty code  (calculate based not the item_difficulty_cuts)
- a pipe delimited string of the item common core standards' codes

### CODE REVIEW

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/StagingToReportingStepsConfig.java [hunk 1]
```diff
@@ -154,7 +154,6 @@ public Step upsertExamsStep() {
             "grade", "completeness", "administration_condition", "ethnicity", "gender",
             "accommodation", "claim", "depth_of_knowledge", "math_practice", "item_trait_score", "language",
             "target", //depends on claim
-            "item_difficulty_cuts", //depends on grade
```
Comment (agorina) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/StagingToReportingStepsConfig.java]: This table is only in the warehouse now.

File: migrate-reporting/src/test/resources/WarehouseEntitiesSetup.sql [hunk 2]
```diff
@@ -43,17 +48,17 @@ INSERT INTO warehouse_test.asmt_score (asmt_id, cut_point_1, cut_point_2, cut_po
...
-  (-992, -99, -99, '200-8906',  -99, -99, -0.03, 2, -99, 1),
-  (-993, -99, -99, '200-2014',  -99, -98,  1.23, 2, -98, 1),
-
-  (-980, -98, -98, '200-60347', -98, -98, -0.32, 1, null, null),
-  (-981, -98, -98, '200-51719', -98, -98, -1.32, 1, null, null),
-  (-982, -98, -98, '200-59217', -98, -98, -2.32, 1, null, null),
-  (-983, -98, -98, '200-59208', -98, -98, -0.32, 1, null, null),
-  (-984, -98, -98, '200-30901', -98, -98, -0.32, 1, null, null);
+INSERT INTO warehouse_test.item (id, claim_id, target_id, natural_id, asmt_id, dok_id, difficulty, difficulty_code, max_points, math_practice, allow_calc) VALUES
+  (-990, -99, -99, '200-2010',  -99, -99, -0.23, 'E',  2, -99, 0),
```
Comment (malaffoon) [inline migrate-reporting/src/test/resources/WarehouseEntitiesSetup.sql]: Is it `E` or `L`?

File: package-processor/src/main/resources/application.sql.yml [hunk 3]
```diff
@@ -56,6 +56,14 @@ sql:
       INSERT INTO item (asmt_id, claim_id, target_id, natural_id, math_practice, allow_calc, dok_id, difficulty, max_points)
                         select (select id as asmt_id from asmt where natural_id=:asmtId), :claimId, :targetId, :itemId, :mathPractice, :allowCalc, :dokId, :difficulty, :maxPoints
 
+    updateDifficultyCode:
+      update item i
+        join asmt a on i.asmt_id = a.id
+        join item_difficulty_cuts c on c.asmt_type_id = a.type_id and a.subject_id = c.subject_id and a.grade_id = c.grade_id
+      set
+        i.difficulty_code = if(i.difficulty < c.moderate_low_end, 'L', if(i.difficulty < c.difficult_low_end, 'M', 'D' ))
```
Comment (malaffoon) [inline package-processor/src/main/resources/application.sql.yml]: This should be `'E'`?

File: migrate-reporting/src/test/resources/WarehouseEntitiesSetup.sql (hunk 2 above)
Comment (agorina) [inline migrate-reporting/src/test/resources/WarehouseEntitiesSetup.sql]: The right way is  **E**asy, **M**oderate, and **D**ifficult.

I have no idea how it turned into 'L'...Thank you for noticing. Will check it everywhere.

**Pull request merged**